README.md

```diff
@@ -20,3 +20,13 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+The runtime complexity of the algorithm that I implemented is $O(n! * n)$. The permutations portion has a complexity of $O(n!)$, this is when it's generating all permutations of an array length n. The isSorted function has a complexity of $O(n)$, which as the name implies checks if the array is sorted or not. Knowing these two complexities I was able to come to the conclusion that the runtime complexity would be $O(n! * n)$. A best case input would be if the array was already sorted. If the array is already sorted then the function will only need to generate and check one permutation. The complexity would then be $O(n)$. A worst case scenerio would occur when the array is reversed. In this case the function would have to check and generate all permutations before finding the sorted one. The complexity would change if you generated permutations randomly without memory instead of systemically trying them. The randomized process could allow for a potentially faster finding of the sorted permutation, since the random could find it in fewer attempts. I believe the random permutation would still have a complexity of $O(n!)$ since each permutation has a 1/n! chance of being in the sorted order. If it was randomized, then there would be a lot of variability to the efficieny. 
```

Comment:
If the permutations are generated completely randomly, is there a guarantee that it would find the solution?

Reply:
If it is generated completely randomly I don't think that there is a guarantee that it would find the solution. I think that if there was not any limitations and if there was an infinite amount of tries then yes, you would eventually be able to find the sorted permutation. However in a practical setting no there is no guarantee.